### PR TITLE
modules/nixos: add nginx

### PR DESCRIPTION
**Pull request by zowoq**
~~Maybe quic as well?~~ Will look at quic in another PR.

~~`net.ipv4.tcp_fastopen=3` ?~~

### CODE REVIEW

File: modules/nixos/nginx.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          '';
+
+          locations."= /robots.txt".alias = pkgs.writeText "robots.txt" ''
+            User-agent: *
+            Disallow: /
+          '';
+        };
+      }
+    );
+  };
```
Comment (zimbatm) [inline modules/nixos/nginx.nix:28]: I didn't know you could re-define existing options

Comment (Mic92) [reply modules/nixos/nginx.nix:28]: Submodules are also modules at the end of the day.

**Comment by Mic92** [comment]
Did you measure the performance of KTLS? I measured it for harmonia and it actually made performance measurable worse.

**Comment by zowoq** [comment]
Sorry, didn't realise that you posted two comments, when I opened the notification I only saw the last one.

No, I didn't check this specific configuration but about a year ago I did a similar nginx config for another project and it seemed fine. Can we try the same measurements here? How different is the config you are using, e.g. worker_processes, quic, reuseport?

**Pull request merged**